### PR TITLE
Reorganize fencer code a little bit

### PR DESCRIPTION
**Pull request by kgaillot**
This was a thread I shouldn't have pulled on :) but it's done now. The main goal is to separate some code into two new source files, fenced_cib.c and fenced_scheduler.c, for code isolation. There is also a bit of best-practices work on the scheduler-related code.

### CODE REVIEW

**Comment by kgaillot** [comment]
@wenningerk , can you review this when you get a chance? Low priority, thanks

File: daemons/fenced/fenced_scheduler.c [hunk 1]
```diff
@@ -97,119 +97,107 @@ local_node_allowed_for(const pcmk_resource_t *rsc)
...
+            if (pe_rsc_is_clone(rsc)) {
+                return; // Only one instance needs to be checked for clones
             }
         }
         return;
     }
 
-    /* We only care about STONITH resources. */
     rclass = crm_element_value(rsc->xml, XML_AGENT_ATTR_CLASS);
     if (!pcmk__str_eq(rclass, PCMK_RESOURCE_CLASS_STONITH, pcmk__str_casei)) {
```
Comment (wenningerk) [inline daemons/fenced/fenced_scheduler.c:135]: Do we need to compare case insensitive here?

**Comment by wenningerk** [comment]
Looking good!

File: daemons/fenced/fenced_scheduler.c (hunk 1 above)
Comment (kgaillot) [reply daemons/fenced/fenced_scheduler.c:135]: We probably *shouldn't*, but I don't want to break any existing configurations. For provider and agent it's easier to insist on case sensitivity, but class names are an invention of Pacemaker. There were probably variations in the early days that might still survive somewhere. We could set a stricter policy at a backward compatibility break.

**Pull request merged**